File: src/main/java/com/example/starwarsunlimiteddb/Utente.java
package com.example.starwarsunlimiteddb;

import java.sql.ResultSet;
import java.sql.SQLException;

public class Utente{
    private String nome;
    private int id;
    private String email;
    private String password;

    public Utente(String nome, int id, String email, String password) {
        this.nome = nome;
        this.id = id;
        this.email = email;
        this.password = password;
    }

    public Utente(ResultSet rs, int i) throws SQLException{
        if(rs.absolute(i)){
            nome = rs.getString("nome");
            id = rs.getInt("idUtente");
            email = rs.getString("email");
            password = rs.getString("password");
        }
    }

    public String getNome(){
        return nome;
    }

    public int getID(){
        return id;
    }

    public String getEmail(){
        return email;
    }

    public String getPassword(){
        return password.replaceAll(".", "*");
    }
}

File: Utente.php
<?php
    class Utente{
        private string $nome;
        private int $id;
        private string $email;
        private string $password;

        function __construct(string $nome, int $id, string $email, string $password){
            $this->nome = $nome;
            $this->id = $id;
            $this->email = $email;
            $this->password = $password;
        }

        public function getNome(): string{
            return $this->nome;
        }
        public function getId(): int{
            return $this->id;
        }
        public function getEmail(): string{
            return $this->email;
        }
        public function getPassword(): string{
            return $this->password;
        }
    }